pygments/lexers/c_like.py

```diff
@@ -663,3 +649,53 @@ class OmgIdlLexer(CLexer):
             include('annotation_appl'),
         ],
     }
+class FlexLexer(CLexer):
+    """
+    Lexer for Flex <http://dinosaur.compilertools.net/flex/index.html>, 
+    A fast scanner generator.
+    .. versionadded:: 2.5
+    """
+    name = 'Flex'
+    aliases = ['flex']
+    filenames = ['*.lex', '*.l']
+    mimetypes = ['text/x-lex']
+
+    tokens = {
+        # root will be for definitions section and user code (through inherit mainly)
+        'root': [
```

Comment:
I would recommend restructuring this lexer. It is using the `CLexer` base class to highlight most things, which means this for example will unavoidably be highlighted incorrectly

```
%%

while { rule_action(); }
```

`while` is a C language keyword, but here it should be recognized as a pattern (`String.Regex`). 

Instead, I would use the trick at https://github.com/pygments/pygments/issues/1893#issuecomment-1111764791 (for lack of something better for the time being; it would be nice ultimately if we had a way to say "start using this other lexer, and hand back to me when it pops all of its stack").

Reply:
So you would suggest to have `FlexLexer` inherit from `RegexLexer` and then use `using()` to handle the C code sections?